conf/defaults.ini

```diff
@@ -835,6 +835,10 @@ max_attempts = 3
 # The interval string is a possibly signed sequence of decimal numbers, followed by a unit suffix (ms, s, m, h, d), e.g. 30s or 1m.
 min_interval = 10s
 
+# Maximum amount of time permitted to take a screenshot of an alert rule before the screenshot is canceled.
+# The interval string is a possibly signed sequence of decimal numbers, followed by a unit suffix (ms, s, m, h, d), e.g. 30s or 1m.
+screenshot_timeout = 15s
+
```

Comment:
Does this mean that we're expecting the rendering + querying to all be done within 15s or just the rendering? I fear there might be plenty of use-cases where a query take > 15s and we wouldn't notify the user that is very likely we can't attach a screenshot. 

I'd be in favour of removing this setting for now and deciding on the next PR once we have some data.

There might be more we can do from the front-end side.

Reply:
Yes, that's right! This timeout is from old alerting.

If querying takes longer than 10 seconds then we can have other issues, such as alert evaluation not being able to keep up with the scheduler (depends on the interval).

Reply:
Yes, that's right! This timeout is from old alerting.

If querying takes longer than 10 seconds then we have other issues, such as alert evaluation not being able to keep up with the scheduler (depends on the interval).

Reply:
Yes, that's right! This timeout is from old alerting.

If querying takes longer than 10 seconds then we have other issues, such as alert evaluation not being able to keep up with the scheduler (depends on the interval).

Reply:
Yes, that's right! This timeout is from old alerting.

If querying takes longer than 10 seconds then we have other issues, such as alert evaluation not being able to keep up with the scheduler (i.e. if the interval is 10s).

Reply:
Yes, that's right! This timeout is from old alerting.

If querying takes longer than 10 seconds then we have other issues, such as alert evaluation not being able to keep up with the scheduler (i.e. if the interval is 10s).

Reply:
How do you feel about removing it for now? I think we need to make a more conscious decision on how we want to document this + and what its default value is going to be further down the line.

Reply:
That would mean no timeout? I think that's going to be worse than a 15 second time?

Reply:
That would mean no timeout? I think that's going to be worse than a 15 second timeout?

Reply:
You use a constant in-code, I meant removing the configuration flag.